File: python/auth/get_access_token.py
import sys
from pathlib import Path
sys.path.append(str(Path(__file__).resolve().parent.parent))

from helpers import send_easylink_request

base_url   = 'https://sandbox.easylink.id'
app_id     = 'lQNJ0nL07Ucmemaa'
app_secret = 'HrfFeuRmoyBsZhxDi3w3JNdxwYu19lL4'

def get_access_token(base_url: str, app_id: str, app_secret: str) -> str:
    response = send_easylink_request(
        base_url,
        '/get-access-token',
        'POST',
        {
            'app_id': app_id,
            'app_secret': app_secret
        },
        '',
        ''
    )

    if response['status_code'] != 200:
        raise Exception(f"Failed to get token: {response}")

    data = response['data']
    
    if isinstance(data.get('data'), str):
        token = data['data']
    else:
        token = data.get('accessToken') or data.get('access_token') or (data.get('data') and (data.get('data').get('accessToken') or data.get('data').get('access_token')))

    if not token:
        raise Exception(f"Access token not found in response: {data}")

    return token

if __name__ == '__main__':
    try:
        print("Requesting Access Token...")
        token = get_access_token(base_url, app_id, app_secret)
        print(f"Access Token retrieved successfully:\n{token}")
    except Exception as e:
        print(f"Error: {e}")

File: nodejs/auth/get_access_token.js
const { sendEasylinkRequest } = require('../helpers');

const baseUrl   = 'https://sandbox.easylink.id';
const appId     = 'YOUR_APP_ID';
const appSecret = 'YOUR_APP_SECRET';

async function getAccessToken(baseUrl, appId, appSecret) {
  const response = await sendEasylinkRequest(
    baseUrl,
    '/get-access-token',
    'POST',
    {
      appId,
      appSecret,
      app_id: appId,
      app_secret: appSecret,
    },
    '',
    ''
  );

  if (response.status !== 200) {
    throw new Error(`Failed to get token: ${JSON.stringify(response)}`);
  }

  const data = response.data;
  let token = null;

  if (typeof data.data === 'string') {
    token = data.data;
  } else {
    token = data.accessToken || data.access_token || (data.data && (data.data.accessToken || data.data.access_token));
  }

  if (!token) {
    throw new Error(`Access token not found in response: ${JSON.stringify(data)}`);
  }

  return token;
}

(async () => {
  try {
    console.log('Requesting Access Token...');
    const token = await getAccessToken(baseUrl, appId, appSecret);
    console.log('Access Token retrieved successfully:\n', token);
  } catch (error) {
    console.error('Error:', error.message);
  }
})();
